services/hanke-service/src/test/kotlin/fi/hel/haitaton/hanke/factory/HakemusyhteystietoBuilder.kt

```diff
@@ -28,65 +28,112 @@ data class HakemusyhteystietoBuilder(
-    ): HakemusyhteystietoBuilder {
-        f(saveYhteystieto(ApplicationContactType.RAKENNUTTAJA, yhteystieto))
-        return this
+        vararg yhteyshenkilot: HankekayttajaEntity =
+            arrayOf(kayttaja(HankeKayttajaFactory.KAYTTAJA_INPUT_RAKENNUTTAJA))
+    ): HakemusyhteystietoEntity {
+        return saveYhteystieto(
+            ApplicationContactType.RAKENNUTTAJA,
+            yhteystieto,
+            yhteyshenkilot.asList()
+        )
     }
 
+    fun rakennuttaja(
+        kayttooikeustaso: Kayttooikeustaso,
+        tilaaja: Boolean = false,
+    ): HakemusyhteystietoEntity {
+        val kayttaja = kayttaja(HankeKayttajaFactory.KAYTTAJA_INPUT_RAKENNUTTAJA, kayttooikeustaso)
+        return saveYhteystieto(ApplicationContactType.RAKENNUTTAJA, kayttaja, tilaaja)
+    }
+
+    fun rakennuttaja(
+        first: HankekayttajaEntity,
+        vararg yhteyshenkilot: HankekayttajaEntity
+    ): HakemusyhteystietoEntity = rakennuttaja(yhteyshenkilot = arrayOf(first) + yhteyshenkilot)
+
     fun asianhoitaja(
-        asianhoitaja: HankekayttajaInput = HankeKayttajaFactory.KAYTTAJA_INPUT_ASIANHOITAJA,
-        kayttooikeustaso: Kayttooikeustaso = Kayttooikeustaso.KATSELUOIKEUS,
         yhteystieto: Hakemusyhteystieto = HakemusyhteystietoFactory.create(),
-        f: (HakemusyhteystietoEntity) -> Unit =
-            defaultYhteyshenkilo(asianhoitaja, kayttooikeustaso),
-    ): HakemusyhteystietoBuilder {
-        f(saveYhteystieto(ApplicationContactType.ASIANHOITAJA, yhteystieto))
-        return this
+        vararg yhteyshenkilot: HankekayttajaEntity =
+            arrayOf(kayttaja(HankeKayttajaFactory.KAYTTAJA_INPUT_ASIANHOITAJA))
+    ): HakemusyhteystietoEntity {
+        return saveYhteystieto(
+            ApplicationContactType.ASIANHOITAJA,
+            yhteystieto,
+            yhteyshenkilot.asList()
+        )
     }
 
-    private fun addYhteyshenkilo(
-        yhteystietoEntity: HakemusyhteystietoEntity,
-        kayttajaInput: HankekayttajaInput,
-        kayttooikeustaso: Kayttooikeustaso = Kayttooikeustaso.KATSELUOIKEUS,
-        tilaaja: Boolean = false
-    ) {
-        val kayttaja =
-            hankeKayttajaFactory.findOrSaveIdentifiedUser(
-                applicationEntity.hanke.id,
-                kayttajaInput,
-                kayttooikeustaso
-            )
-        addYhteyshenkilo(yhteystietoEntity, kayttaja, tilaaja)
+    fun asianhoitaja(
+        kayttooikeustaso: Kayttooikeustaso,
+        tilaaja: Boolean = false,
+    ): HakemusyhteystietoEntity {
+        val kayttaja = kayttaja(HankeKayttajaFactory.KAYTTAJA_INPUT_ASIANHOITAJA, kayttooikeustaso)
+        return saveYhteystieto(ApplicationContactType.ASIANHOITAJA, kayttaja, tilaaja)
     }
 
+    fun asianhoitaja(
+        first: HankekayttajaEntity,
+        vararg yhteyshenkilot: HankekayttajaEntity
+    ): HakemusyhteystietoEntity = rakennuttaja(yhteyshenkilot = arrayOf(first) + yhteyshenkilot)
```

Comment:
Should it be `HakemusyhteystietoEntity = asianhoitaja(..)` instead?

Reply:
Thank you. Fixed.